pkg/monitor/monitor/util.go

```diff
@@ -505,14 +512,14 @@ func getMonitorGrafanaContainer(secret *core.Secret, monitor *v1alpha1.TidbMonit
 			},
 		},
 	}
+	if monitor.Spec.Grafana.ImagePullPolicy != nil {
+		c.ImagePullPolicy = *monitor.Spec.Grafana.ImagePullPolicy
+	}
 	for k, v := range monitor.Spec.Grafana.Envs {
-		c.Env = append(c.Env, core.EnvVar{
+		util.AppendOverwriteEnv(c.Env, []core.EnvVar{{
 			Name:  k,
 			Value: v,
-		})
-	}
-	if monitor.Spec.Grafana.ImagePullPolicy != nil {
-		c.ImagePullPolicy = *monitor.Spec.Grafana.ImagePullPolicy
+		}})
```

Comment:
`c.Env = util.AppendOverwriteEnv(c.Env, monitor.Spec.Grafana.Envs)`

Reply:
These types are not compatible, so I turned it into a loop.